core/embed/projects/prodtest/cmd/prodtest_secrets.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <trezor_model.h>
+
+#include <string.h>
+
+#include <rtl/cli.h>
+#ifdef USE_OPTIGA
+#include <sec/optiga_commands.h>
+#endif
+#include <sec/secret.h>
+#include <sec/secret_keys.h>
+
+#include "memzero.h"
+#include "rand.h"
+#include "secbool.h"
+
+secbool generate_random_secret(uint8_t* secret, size_t length) {
+  random_buffer(secret, length);
+
+#ifdef USE_OPTIGA
+  uint8_t optiga_secret[length];
+  if (OPTIGA_SUCCESS != optiga_get_random(secret, length)) {
+    return secfalse;
+  }
+  for (size_t i = 0; i < length; i++) {
+    secret[i] ^= optiga_secret[i];
+  }
+#endif
```

Comment:
1. bug: optiga_get_random overwrites secret
2. memzero optiga_secret  when done or get rid of the buffer like this:
```suggestion
#ifdef USE_OPTIGA
  if (OPTIGA_SUCCESS != optiga_get_random(secret, length)) {
    return secfalse;
  }
#endif
random_xor(secret, length);
```

Reply:
Fixed in https://github.com/trezor/trezor-firmware/pull/5281/commits/ec838b04cee84e0d8d2dac92ddfca0b519733364.